# Fisher's exact test
* Use to determine if there is a significant association between 2 categorical variables.
* Use in small sample size **($n < 20$)** or if one of the cell in 2x2 table is **less than 5**.
* This test is an alternative to Chi Square test of indepence. 

# Is there an association between being overweight and being a good singer?
If you think about it, the respiratory system if the primary tool to sing. The diaphragm pulls air into your lungs, your abdominal and rib cage muscles pushes the the air out. The vibration in your vocal chords produce the sounds.
<br>
Now, we go to being overweight or obese as a factor because the large amount of fat in the abdominal area exerts more pull in addition to gravity, which may train the abdominal muscles to be stronger. And, this will eventually help you to be a good singer (in addition to training).
<br>
After all, we have famous singers like Luciano Pavarotti, Black Jack, Adele, Jennifer Hudson, etc.
<br>
So you enroll about 16 people in your study. You compute their BMI and you do a "blind audition" using professional singers as judge whether the study subjects are good singers.


In [1]:
# Import the libraries

import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

In [2]:
# Create your dataset in 2x2 table

sing = pd.DataFrame({'Good singer': [4, 2], 'Bad singer': [1, 9]},
                    index=['overweight/obese', 'normal BMI'])
sing

Good singer  Bad singer
overweight/obese            4           1
normal BMI                  2           9

Formulate your hypothesis
<br>
$H_0$: Weight (or BMI) and singing skills are independent. They have no association whatsoever.
<br>
$H_1$: Weight (or BMI) and singing skills are not independent. They have an association with each other.
<br>
Set your level of significance at $\alpha = 0.05$

In [3]:
# Perform the Fisher exact test

Fisher, pval = stats.fisher_exact(sing)
Fisher, pval

(18.0, 0.03571428571428579)

Base on the p-value of 0.0357, which smaller than our $\alpha$ level, we can reject our null hypothesis. There is an association between weight and singing skills.

# Is there an association between probiotics use and gingivitis?
There are many risk factors and causes of gingivitis one which is bacterial invasion of the gingival tissue. Some of these bacteria are: Fusobacterium, Lachnopiraceae, Lautropia, Prevotella, and Rothia. What if we just substitute these bacteria with "good bacteria" like Lactobacillus, Bifidobacterium, and Streptococcus? Will that decrease our chances of having gingivitis in the first place?
<br> <br>
You then enroll 20 subjects, you assigned half of them to taking priobiotics everyday, and the other half to their usual diet. At the end of 1 month, you check them for gingivitis.

In [4]:
# Create your dataset in 2x2 table

probiotics = pd.DataFrame({'With Probiotics': [7, 3], 'Without Probiotics': [8, 2]}, 
                          index=['Healthy gums', 'with gingivitis'])
probiotics

With Probiotics  Without Probiotics
Healthy gums                   7                   8
with gingivitis                3                   2

Formulate your hypothesis
<br>
$H_0$: Probiotics use and gingivitis are independent. They have no association whatsoever.
<br>
$H_1$: Probiotics use and gingivitis are not independent. They have an association with each other.
<br>
Set your level of significance at $\alpha = 0.05$

In [5]:
# Perform the Fisher exact test

Fisher, pval = stats.fisher_exact(probiotics)
Fisher, pval

(0.5833333333333334, 1.0)

We have a p-value of 1.0 which is greater than our $\alpha$. We cannot reject our null hypothesis. There is no association with probiotics use and gingival health.

Note: These are very simplified examples for STEM high school students. It can lead to misinterpretation of the variables that is being compared. Use the above examples with caution.